### PR TITLE
fix bug merge

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 1]
```diff
@@ -634,15 +642,23 @@ public void validateDiffMergeRequest(
...
+        List<String> parentSHAs = Arrays.asList(internalCommitA.getCommit_hash(), internalCommitB.getCommit_hash());
         List<CommitEntity> parentCommits = Arrays.asList(internalCommitA, internalCommitB);
         String mergeMessage = request.getContent().getMessage();
         if (mergeMessage.isEmpty()) {
-          mergeMessage =
-              "Merge "
-                  + request.getCommitShaA().substring(0, 7)
-                  + " into "
-                  + request.getCommitShaB().substring(0, 7);
+          if (branchOrCommitA.length() > 7) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: if it is a branch name then do not truncate the string

**Review by ravishetye — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 2]
```diff
@@ -548,46 +550,50 @@ public void validateDiffMergeRequest(
...
         BranchEntity branchBEntity =
             repositoryDAO.getBranchEntity(
                 readSession, repositoryEntity.getId(), request.getBranchB());
         internalCommitB = readSession.get(CommitEntity.class, branchBEntity.getCommit_hash());
       }
 
       if (!request.getCommitShaA().isEmpty()) {
-        errorNameA = request.getCommitShaA();
+        LOGGER.debug("Commit A found in request");
+        branchOrCommitA = request.getCommitShaA();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: you can do the substring here, and on the other commit

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: if youtruncate the commit sha on top then you can just. have the code you have in else

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: Fixed

**Pull request merged**